File: native-android/app/src/main/java/com/iganapolsky/randomtimer/analytics/PostHogAnalyticsConfigFactory.kt
package com.iganapolsky.randomtimer.analytics

import com.posthog.android.PostHogAndroidConfig

/** Builds PostHog Android SDK config shared by [AnalyticsService]. */
object PostHogAnalyticsConfigFactory {
    private const val POSTHOG_HOST = "https://us.i.posthog.com"

    fun create(
        apiKey: String,
        isInternalUser: Boolean,
    ): PostHogAndroidConfig =
        PostHogAndroidConfig(
            apiKey = apiKey,
            host = POSTHOG_HOST,
        ).apply {
            captureApplicationLifecycleEvents = false
            captureDeepLinks = true
            captureScreenViews = false
            preloadFeatureFlags = true
            errorTrackingConfig.autoCapture = !isInternalUser
            if (!isInternalUser) {
                sessionReplay = true
                sessionReplayConfig.maskAllTextInputs = true
                sessionReplayConfig.maskAllImages = true
                sessionReplayConfig.captureLogcat = true
                sessionReplayConfig.screenshot = true
                sessionReplayConfig.debouncerDelayMs = 1000L
            }
        }
}

File: native-ios/RandomTimer/Sources/Services/PostHogAnalyticsConfigFactory.swift
import Foundation
#if canImport(PostHog)
import PostHog
#endif

/// Builds PostHog iOS SDK config shared by `AnalyticsService` (parity with Android `PostHogAnalyticsConfigFactory`).
enum PostHogAnalyticsConfigFactory {
    private static let posthogHost = "https://us.i.posthog.com"

#if canImport(PostHog)
    static func make(apiKey: String, isInternalUser: Bool) -> PostHogConfig {
        let config = PostHogConfig(apiKey: apiKey, host: posthogHost)
        config.captureApplicationLifecycleEvents = false
        config.captureScreenViews = false
        config.errorTrackingConfig.autoCapture = !isInternalUser
        if !isInternalUser {
            config.sessionReplay = true
            config.sessionReplayConfig.maskAllTextInputs = true
            config.sessionReplayConfig.maskAllImages = true
            config.sessionReplayConfig.captureLogs = true
            config.sessionReplayConfig.captureNetworkTelemetry = true
            config.sessionReplayConfig.screenshotMode = true
            config.sessionReplayConfig.throttleDelay = 1.0
        }
        return config
    }
#endif
}
